channels/tasks.py

```diff
@@ -514,3 +525,112 @@ def check_comment_for_spam(*, user_ip, user_agent, comment_id):
     )
     if is_spam:
         admin_api.remove_comment(comment.comment_id)
+
+
+@app.task
+def update_spam(*, spam, comment_ids, post_ids, retire_users):
+    """
+    Updates the spam check result for comments and posts
+
+    Args:
+        spam (bool): Mark as spam if true and ham if false
+        comment_ids( list of int): list of comment ids in integer form
+        post_ids( list of int): list of post ids in integer for
+        retire_users(bool): retire comment/post authors if true
+
+    """
+    comment_content_type = ContentType.objects.get_for_model(Comment)
+    post_content_type = ContentType.objects.get_for_model(Post)
+    admin_api = get_admin_api()
+
+    akismet_client = create_akismet_client()
+
+    for comment in Comment.objects.filter(id__in=comment_ids).iterator():
+        result = SpamCheckResult.objects.get(
+            content_type=comment_content_type, object_id=comment.id
+        )
+
+        if spam:
+            result.is_spam = True
+            result.save()
+
+            admin_api.remove_comment(comment.comment_id)
+
+            akismet_client.submit_spam(
+                user_agent=result.user_agent,
+                user_ip=result.user_ip,
+                comment_content=comment.text,
+                comment_type="reply",
+                comment_author=comment.author.profile.name,
+                comment_author_email=comment.author.email,
+            )
+
+            if retire_users:
+                retire_user(comment.author)
+
+        else:
+            result.is_spam = False
+            result.save()
+
+            admin_api.approve_comment(comment.comment_id)
+
+            akismet_client.submit_ham(
+                user_agent=result.user_agent,
+                user_ip=result.user_ip,
+                comment_content=comment.text,
+                comment_type="reply",
+                comment_author=comment.author.profile.name,
+                comment_author_email=comment.author.email,
+            )
+
+    for post in Post.objects.filter(id__in=post_ids).iterator():
+        result = SpamCheckResult.objects.get(
+            content_type=post_content_type, object_id=post.id
+        )
+
+        if spam:
+            result.is_spam = True
+            result.save()
+
+            admin_api.remove_post(post.post_id)
+
+            akismet_client.submit_spam(
+                user_agent=result.user_agent,
+                user_ip=result.user_ip,
+                comment_content=post.plain_text,
+                comment_type="forum-post",
+                comment_author=post.author.profile.name,
+                comment_author_email=post.author.email,
+            )
+
+            if retire_users:
+                retire_user(post.author)
+
+        else:
+            result.is_spam = False
+            result.save()
+
+            admin_api.approve_post(post.post_id)
+
+            akismet_client.submit_ham(
```

Comment:
Since `submit_ham` and `submit_spam` both use the same arguments, I think you could decrease the amount of code by doing something like this for both posts and comments:

```python
if spam:
     submit_func = akismet_client.submit_spam
     ....etc
else:
    submit_func = akismet_client.submit_ham
    ....etc
submit_func(
    user_agent=result.user_agent,
    user_ip=result.user_ip,
    ....etc
)
```